pkg/controller/v1alpha1/inferencegraph/knative_reconciler.go

```diff
@@ -92,12 +92,28 @@ func (r *GraphKnativeServiceReconciler) Reconcile(ctx context.Context) (*knservi
 	desired := r.Service
 	existing := &knservingv1.Service{}
 
+	forceStopRuntime := false
+	if val, exist := desired.Spec.Template.Annotations[constants.StopAnnotationKey]; exist {
+		forceStopRuntime = strings.EqualFold(val, "true")
+	}
+
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
 		log.Info("Updating inference graph knative service", "namespace", desired.Namespace, "name", desired.Name)
 		if err := r.client.Get(ctx, types.NamespacedName{Name: desired.Name, Namespace: desired.Namespace}, existing); err != nil {
 			return err
 		}
 
+		if forceStopRuntime {
+			log.Info("Deleting inference graph knative service", "namespace", existing.Namespace, "name", existing.Name)
+			if existing.GetDeletionTimestamp() == nil { // check if the ksvc was already deleted
+				err := r.client.Delete(ctx, existing)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
```

Comment:
The current logic immediately returns even if the Knative service is still in the process of being deleted (DeletionTimestamp != nil).

This might lead to a premature exit, skipping further reconciliation until the next trigger. I think a better approach would be to requeue the request when deletion is in progress. Also, adding a log message when deletion is still pending would help with observability and debugging.